### PR TITLE
Better document end reactive routes

### PR DESCRIPTION
**Pull request by loicmathieu**

### CODE REVIEW

File: extensions/vertx-web/runtime/src/main/java/io/quarkus/vertx/web/RoutingExchange.java [hunk 1]
```diff
@@ -29,30 +35,52 @@ default HttpServerRequest request() {
...
+     * Set the response status code to 200, write a chunck of data to the response then ends it.
+     * @param chunk
+     */
     default void ok(String chunk) {
         ok().end(chunk);
     }
 
+    /**
+     * Set the response status code to 200 and return the response.
+     * YYou must call <code>HttpServerResponse.end()</code> afterwards to end the response.
```
Comment (cihati) [inline extensions/vertx-web/runtime/src/main/java/io/quarkus/vertx/web/RoutingExchange.java:72]: You

File: extensions/vertx-web/runtime/src/main/java/io/quarkus/vertx/web/RoutingExchange.java [hunk 2]
```diff
@@ -29,30 +35,52 @@ default HttpServerRequest request() {
...
+     * Set the response status code to 200 and return the response.
+     * You must call <code>HttpServerResponse.end()</code> afterwards to end the response.
+     * @return the HTTP response object
+     */
     default HttpServerResponse ok() {
         return response().setStatusCode(200);
     }
 
+    /**
+     * Set the response status code to 200, write a chunck of data to the response then ends it.
```
Comment (cihati) [inline extensions/vertx-web/runtime/src/main/java/io/quarkus/vertx/web/RoutingExchange.java:63]: chunk

File: docs/src/main/asciidoc/reactive-routes.adoc [hunk 3]
```diff
@@ -200,6 +200,9 @@ A route method must be a non-private non-static method of a CDI bean.
 If the annotated method returns `void` then it has to accept at least one argument - see the supported types below. 
 If the annotated method does not return `void` then the arguments are optional.
 
+NOTE: Methods that return `void` must __end__ the response or the HTTP request to this route will never ends.
```
Comment (gsmet) [inline docs/src/main/asciidoc/reactive-routes.adoc:203]:
```suggestion
NOTE: Methods that return `void` must __end__ the response or the HTTP request to this route will never end.
```

File: docs/src/main/asciidoc/reactive-routes.adoc [hunk 4]
```diff
@@ -200,6 +200,9 @@ A route method must be a non-private non-static method of a CDI bean.
...
+Some methods of the `RoutingExchange` do it for you, others not and you must call the `end()` method of the response by yourself, please refer to its JavaDoc for more information.
```
Comment (gsmet) [inline docs/src/main/asciidoc/reactive-routes.adoc:204]:
```suggestion
Some methods of `RoutingExchange` do it for you, others not and you must call the `end()` method of the response by yourself, please refer to its JavaDoc for more information.
```

**Review by gsmet — DISMISSED** [review]
Spotted a small typo, I will let @mkouba review this.

File: extensions/vertx-web/runtime/src/main/java/io/quarkus/vertx/web/RoutingExchange.java (hunk 1 above)
Comment (loicmathieu) [reply extensions/vertx-web/runtime/src/main/java/io/quarkus/vertx/web/RoutingExchange.java]: Corrected

File: extensions/vertx-web/runtime/src/main/java/io/quarkus/vertx/web/RoutingExchange.java (hunk 2 above)
Comment (loicmathieu) [reply extensions/vertx-web/runtime/src/main/java/io/quarkus/vertx/web/RoutingExchange.java]: Corrected

**Review by mkouba — APPROVED** [review]
Looks good. Thanks!

**Comment by mkouba** [comment]
@loicmathieu It seems that we need to format the `RoutingExchange`, i.e. just run `mvn clean install` in `quarkus/extensions/vertx-web`...
```
Caused by: org.apache.maven.plugin.MojoFailureException: File '/home/runner/work/quarkus/quarkus/extensions/vertx-web/runtime/src/main/java/io/quarkus/vertx/web/RoutingExchange.java' has not been previously formatted
```

**Comment by loicmathieu** [comment]
@mkouba sorry, forgot to launch a build to format the class. 

Done, should be good now

**Pull request merged**